### PR TITLE
Replace current theme components with `@swmansion/t-rex-ui`

### PR DESCRIPTION
**Pull request by patrycjakalinska**
## Description

Because of a ramping up maintainance cost across our suite of tools - Reanimated, Gesture Handler & React Native IDE we've decided to create a UI library for our reusable docusaurus components.

This PR replaces old theme components such as Admonition, Sidebar and Table of Contents with new, reusable ones from `@swmansion/t-rex-ui`.

Full list of replaced components:
* Admonition
* PaginatorNavLink
* DocSidebar
* DocSidebarItem
* DocSidebarItems
* DocCard
* DocItemTOCMobile
* DocItemMetadata
* TOCCollapsible
* TOCItems
* TOCItemTree
* Footer
* Navbar
* SearchPage

## Test plan

```
cd docs/
yarn

# to test development build
yarn start

# to test production build
yarn build
yarn serve
```

### CODE REVIEW

File: packages/docs/src/css/overrides.css [hunk 1]
```diff
@@ -97,11 +98,25 @@ table thead tr {
...
 }
 
 [class*="navbar__item"] {
   display: flex !important;
 }
+
+/* TODO: probably to remove */
+[class*="logoWrapper"] [class*="sidebarLogo"] {
+  margin-left: 1.5rem;
+}
```
Comment (m-bert) [inline packages/docs/src/css/overrides.css:113]: If it is _"probably to remove"_, why not just remove it?

File: packages/docs/src/theme/DocSidebar/index.js [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
+export default function DocSidebarWrapper(props) {
+  const heroImages = {
+    logo: useBaseUrl("/img/logo.svg"),
+  };
   return (
     <>
-      {shouldRenderSidebarDesktop && <DocSidebarDesktop {...props} />}
-      {shouldRenderSidebarMobile && <DocSidebarMobile {...props} />}
+      <DocSidebar heroImages={heroImages} {...props} />
     </>
```
Comment (m-bert) [inline packages/docs/src/theme/DocSidebar/index.js:12]: That `Fragment` seems unnecessary.

**Review by m-bert — COMMENTED** [review]
Looks good!

File: packages/docs/src/css/overrides.css (hunk 1 above)
Comment (patrycjakalinska) [reply packages/docs/src/css/overrides.css:113]: good thinking, removed 👍

**Review by m-bert — APPROVED** [review]
:shipit:

**Pull request merged**